RecoLocalCalo/EcalRecAlgos/src/EcalUncalibRecHitMultiFitAlgo.cc

```diff
@@ -114,7 +111,8 @@ EcalUncalibratedRecHit EcalUncalibRecHitMultiFitAlgo::makeRecHit(const EcalDataF
   //optionally apply a stricter criteria, assuming slew rate limit is only reached in case where maximum sample has gain switched but previous sample has not
   //option 1: use simple max-sample algorithm
   if (hasGainSwitch && _gainSwitchUseMaxSample) {
-    EcalUncalibratedRecHit rh( dataFrame.id(), maxamplitude, pedval, 0., 0., flags );
+    double maxpulseamplitude = maxamplitude / fullpulse[iFullPulseMax];
```

Comment:
Yes it's non-zero by construction (or at least this would be a really pathological entry in the conditions db).  In all existing tags it's 1.0, and even if this changes in the future, it would have a pretty narrow range of values around 1.0 (+- a few percent at most)